ReSharper.FSharp/src/FSharp.Psi/src/FSharp.psi

```diff
@@ -1083,6 +1083,10 @@ unionCaseDeclaration options { stubBase="FSharpTypeElementDeclarationBase"; } ex
   attributeList<ATTRIBUTE_LIST, AttributeLists>*
   accessModifier{ACCESS_MODIFIER, AccessModifier}?
   identifier<IDENTIFIER, Identifier>
+  (
+    COLON<FSHARP_COLON, Colon>
+    typeUsage<TYPE, TypeUsage>
+  )?
```

Comment:
It looks like it should be either a full type or a field list. Could you try reflecting that in the grammar, please?